### DISCUSSION TITLE
Definition of logical relation after adding a new environment delta

### DISCUSSION THREAD

**Discussion by Bohun9** [ppolesiuk/type-systems-notes#55] [Q&A] [answered]
It is not clear to me what is the definition of $\Gamma; \Delta \models e : \tau$.

**Comment by ppolesiuk** [comment, marked as answer]
Yes, this is not typeset yet. As for simple types we need to quantify over all semantic objects that represents environments (Γ and Δ in case of polymorphism). In case of environment for type variables (Δ, which is just a set), the semantic representation is a function η: Δ → **Type**, that for each type variable returns its meaning in terms of semantic types. Therefore, we have

Γ; Δ ⊨ e : τ  ⇔  ∀ η : Δ → **Type**. ∀ γ ∈ ⟦ Γ ⟧η. γ* e ∈ E⟦ τ ⟧η.
